Mage/src/main/java/mage/abilities/effects/common/ExileCardsFromTopOfLibraryTargetEffect.java

```diff
@@ -1,39 +1,37 @@
 package mage.abilities.effects.common;
 
 import mage.abilities.Ability;
+import mage.abilities.dynamicvalue.DynamicValue;
+import mage.abilities.dynamicvalue.common.StaticValue;
 import mage.abilities.effects.OneShotEffect;
 import mage.cards.Cards;
 import mage.cards.CardsImpl;
 import mage.constants.Outcome;
 import mage.constants.Zone;
 import mage.game.Game;
 import mage.players.Player;
-import mage.util.CardUtil;
 
 /**
- * @author LevelX2
+ * @author LevelX2, Susucr
  */
 public class ExileCardsFromTopOfLibraryTargetEffect extends OneShotEffect {
 
-    int amount;
-    String targetName;
+    private final DynamicValue amount;
 
-    public ExileCardsFromTopOfLibraryTargetEffect(int amount) {
-        this(amount, null);
+    public ExileCardsFromTopOfLibraryTargetEffect(int amount, String targetName) {
+        this(StaticValue.get(amount), targetName);
     }
 
-    public ExileCardsFromTopOfLibraryTargetEffect(int amount, String targetName) {
+    public ExileCardsFromTopOfLibraryTargetEffect(DynamicValue amount, String targetName) {
         super(Outcome.Exile);
-        this.amount = amount;
-        this.staticText = (targetName == null ? "that player" : targetName) + " exiles the top "
-                + CardUtil.numberToText(amount, "")
-                + (amount == 1 ? "card" : " cards") + " of their library";
+        this.amount = amount.copy();
+        this.staticText = targetName + " exiles the top " + amount + " card"
```

Comment:
Just realized that the target name should really be set from the target pointer by overriding getText(Mode mode) rather than passing in a constructor parameter

Reply:
just something like that?
```
    @Override
    public String getText(Mode mode) {
        if (staticText != null && !staticText.isEmpty()) {
            return staticText;
        }
        return getTargetPointer().describeTargets(mode.getTargets(), "that player")
                + " exiles the top " + amount + " card" + (amount.toString().equals("1") ? " " : "s ")
                + " of their library";
    }
```
Checking the 5 cards Rules text with that.

Reply:
Ok needed a tweak to make the text version of amount, and handle 1 better, but now there is no longer that targetName in constructors.